feat(lint): add support for GitHub Actions output format (#1497)
Redocly/redocly-cli: .changeset/tame-gifts-tease.md

---
"@redocly/openapi-core": patch

> "@redocly/cli": patch

Reviewer: Would we classify this as a patch or feature?

It seems like a new feature to me and would result in a new minor version?

I didn't review the substance of the PR but the spirit is awesome. 🏆 

Thank you @rjelierse

Author: I was debating whether this should be considered a new function, but new minor function works for me.

---

> Add support for `github-actions` as an output format for the `redocly lint` command. This will annotate reported problems on files when used in a workflow on GitHub Actions.

Reviewer 2: ```suggestion
Added support for a `github-actions` output format for the `lint` command to annotate reported problems on files when used in a GitHub Actions workflow.
```
